### PR TITLE
zoom in/out animation with css transitions

### LINKED ISSUE #1220: Implement zoom in / out animation with a CSS transition
Got blocked by https://github.com/servo/servo/issues/12749

### PR DESCRIPTION
**Pull request by Gozala**
# Do not land, blocked by https://github.com/servo/servo/issues/12749

Fixes #1220

### CODE REVIEW

**Comment by paulrouget** [comment]
I believe this is better because for performance, but can you explain why we want to do that now?

**Comment by tschneidereit** [comment]
> I believe this is better because for performance, but can you explain why we want to do that now?

We had talked about this in May/June and postponed it until after the demo and Nightly. @pcwalton thought it's quite important to do this, and I can easily believe that to be true.

@Gozala, looks like the blocker has been fixed, so I guess we can go forward with this now?

File: src/browser/Navigators.js [hunk 1]
```diff
@@ -507,6 +427,22 @@ const styleSheet = Style.createSheet
       , overflow: 'hidden'
       , transformOrigin: 'left center'
       , boxSizing: 'border-box'
+      , transition: `border-right 200ms ${easeOutCubic}, transform 200ms ${easeOutCubic}`
+      , background: 'white'
+      }
+    , shrinked: {
```
Comment (tschneidereit) [inline src/browser/Navigators.js]: To be proper English, this would need to be `shrunken`

File: src/browser/Navigators.js [hunk 2]
```diff
@@ -507,6 +427,22 @@ const styleSheet = Style.createSheet
...
+        borderRight: 'solid transparent 50px'
+      }
+    , expanded: {
+        borderRight: 'solid transparent 0px'
+      }
+    , collapsed:
+      { transform: 'translate3d(0, 0, 0)'
+      , transition: `border-right 200ms ${easeOutCubic}, transform 200ms ${easeOutCubic}`
```
Comment (tschneidereit) [inline src/browser/Navigators.js]: Won't this be copied from `base` and thus already exist?

File: src/browser/Navigators.js [hunk 3]
```diff
@@ -507,6 +427,22 @@ const styleSheet = Style.createSheet
...
+      }
+    , expose:
+      { transition: `border-right 200ms ${easeOutCubic}, transform 500ms ${easeOutCubic}`
```
Comment (tschneidereit) [inline src/browser/Navigators.js]: Same here.

**Comment by Gozala** [comment]
Closing this for now.

**Pull request closed without merging**